### PR TITLE
chore: Add release notes for Mender 3.7.1 (C++ client)

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: 20.Mender-client/docs.md [hunk 1]
```diff
@@ -7,6 +7,152 @@ shortcode-core:
...
+  ([MEN-6671](https://northerntech.atlassian.net/browse/MEN-6671))
+* The client no longer erronously commits a rootfs-image
+  artifact after being restarted using `systemctl restart` in the
+  `ArtifactReboot` state.
+  ([MEN-6633](https://northerntech.atlassian.net/browse/MEN-6633))
+
+##### Features
+
+* build and test using the latest version of golang (1.21 today)
+  ([QA-614](https://northerntech.atlassian.net/browse/QA-614))
```
Comment (tranchitella) [inline 20.Mender-client/docs.md:106]: This doesn't make too much sense

File: 20.Mender-client/docs.md [hunk 2]
```diff
@@ -7,6 +7,152 @@ shortcode-core:
...
+| Northern.tech | 13 (81.2%) |
+| aduskett@gmail.com | 1 (6.2%) |
+| Tronel | 1 (6.2%) |
+| BlackBerry Limited | 1 (6.2%) |
+
+### Changelogs
+
+#### mender (4.0.0)
+
+New changes in mender since 3.5.2:
```
Comment (tranchitella) [inline 20.Mender-client/docs.md:86]: No mentions this is the "big rewrite project"?

File: 20.Mender-client/docs.md (hunk 1 above)
Comment (lluiscampos) [reply 20.Mender-client/docs.md:106]: It doesn't make sense anymore.

It seems like [this change](https://github.com/mendersoftware/mender/commit/81f5e35172fc08db04120db8dd6669d76c8e9ebd) was marked as feature and we never ported it to 3.5.x. Hence it never got released it.

I will manually remove this entry.

File: 20.Mender-client/docs.md (hunk 2 above)
Comment (lluiscampos) [reply 20.Mender-client/docs.md:86]: I can add here some note pointing to the mender-docs page that explains the upgrade and split. It depends on:
* https://github.com/mendersoftware/mender-docs/pull/2289

Comment (lluiscampos) [reply 20.Mender-client/docs.md:86]: I can add here some note pointing to the mender-docs page that explains the upgrade and split. It depends on:
* https://github.com/mendersoftware/mender-docs/pull/2289

Comment (kacf) [reply 20.Mender-client/docs.md:86]: Ah yes, I was planning to make a bigger note about it. If you want, I can take over this PR, and combine with my work in the other PR. Just assign it to me.

Comment (lluiscampos) [reply 20.Mender-client/docs.md:86]: Thank you :pray:

Comment (kacf) [reply 20.Mender-client/docs.md:92]: Mentioned the rewrite at the top level, with a link to the documentation for the migration guide.

**Comment by kacf** [comment]
@lluiscampos: I can't request review from you, since it's your PR, but please take a look!

File: 20.Mender-client/docs.md [hunk 3]
```diff
@@ -7,6 +7,156 @@ shortcode-core:
...
+| aduskett@gmail.com | 1 (6.2%) |
+| Tronel | 1 (6.2%) |
+| BlackBerry Limited | 1 (6.2%) |
+
+### Changelogs
+
+#### mender (4.0.0)
+
+Mender client version 4.0.0 has been completely rewritten from scratch in C++. This comes with a
+number of changes which are important to pay attention to to ensure a successful migration. The
```
Comment (lluiscampos) [inline 20.Mender-client/docs.md:87]:
```suggestion
Mender client version 4.0.0 has been completely rewritten from scratch in C++. To ensure a successful migration, there are a number of changes which are important to pay attention to. The
```

**Review by lluiscampos — COMMENTED** [review]
@kacf I cannot approve it neither because it is my PR :)

See below for a rewording suggestion.

**Pull request merged**